logos/logos-agent/app/sessions.py

```
@@ -1493,7 +1505,46 @@ async def _may_try_again(session: dict[str, Any]) -> bool:
+        told it, and the next thing that happened was a person finding a red
+        pull request.
+
+        Watched in the background because it takes minutes: the session is
+        settled, its workspace is free, and what comes of this is a fresh
+        attempt with the failure in its task.
+        """
+        branch = str(session.get("branch_name") or "")
+        if not branch or not session.get("trigger_ref"):
+            # A session queued by a person is that person's to follow up.
+            return
+        asyncio.create_task(self._take_up_a_red_build(session, branch, pushed_sha))
```

Ruff (0.16.3)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)